src/combined-selectors/reporting-config-search-results.ts

```diff
@@ -54,16 +55,100 @@ function searchReportingConfig(
 		}
 	}
 
+	// Finally, search for the search term in the description of the products and feature areas
+	const searchTermTokens = tokenizeAndNormalize( searchTerm );
+	const scores: Record< string, number > = {};
+
+	for ( const token of searchTermTokens ) {
+		const matchingEntities = invertedIndex.get( token ) || [];
+
+		for ( const { id, weight } of matchingEntities ) {
+			if ( ! scores[ id ] ) {
+				scores[ id ] = 0;
+			}
+			scores[ id ] += weight;
+		}
+	}
+
+	const scoreThreshold = 1;
+
+	for ( const entityId in scores ) {
+		if ( scores[ entityId ] >= scoreThreshold ) {
+			if ( products[ entityId ] ) {
+				searchResults.products.add( entityId );
+			} else if ( featureGroups[ entityId ] ) {
+				searchResults.featureGroups.add( entityId );
+				searchResults.products.add( featureGroups[ entityId ].productId );
+			} else if ( features[ entityId ] ) {
+				searchResults.features.add( entityId );
+				if ( features[ entityId ].parentType === 'product' ) {
+					searchResults.products.add( features[ entityId ].parentId );
+				} else {
+					searchResults.featureGroups.add( features[ entityId ].parentId );
+					const parentFeatureGroup = featureGroups[ features[ entityId ].parentId ];
+					searchResults.products.add( parentFeatureGroup.productId );
+				}
+			}
+		}
+	}
+
 	return searchResults;
 }
 
+function addTokensToInvertedIndex(
+	entityId: string,
+	description: string | undefined,
+	invertedIndex: Map< string, Array< { id: string; weight: number } > >
+) {
+	const descriptionTokens = description ? tokenizeAndNormalize( description ) : [];
+	const tokensWithWeights = [ ...descriptionTokens.map( ( token ) => ( { token, weight: 1 } ) ) ];
+
+	for ( const { token, weight } of tokensWithWeights ) {
+		if ( ! invertedIndex.has( token ) ) {
+			invertedIndex.set( token, [] );
+		}
+
+		const tokenList = invertedIndex.get( token );
+		if ( tokenList ) {
+			tokenList.push( { id: entityId, weight } );
+		}
+	}
```

Comment:
**Suggestion (blocking)**: (Qualifier: I feel bad because I never documented this assumption anywhere, I'm sorry! 🙇 )

But, to be safe, I think we should avoid assuming that every entity ID is unique across the board. I realize this is currently how things work, but it's something I could see changing in the future. However, we _can_ assume that every entity ID is unique _within_ a given entity map (products, for example) -- that should never change.

All that to say... Would you be okay with storing the id and the entity type here in the index?

Then above, when you are searching the inverted index, you would just check the entity type directly on the result instead of checking for the entity ID in each entity map.

What do you think?

Reply:
Oohhh, that makes sense and your suggestion is a good idea! I hope I understood it correctly. The changes can be found here: [109f56b](https://github.com/Automattic/bugomattic/pull/80/commits/109f56bdd44747aabe246ea198a9cd8c25c27d8f)